### PR TITLE
[Java] Update PUT excluding fields

### PR DESCRIPTION
**Pull request by KavithaSiva**
## What Has Changed?

I have added information about using `excludingFields` during PUT Update request.
I have also included the [improvement shipped by Matthias](https://github.wdf.sap.corp/MA/sdk/pull/6746) on this method.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

**Review by MatKuhr — COMMENTED** [review]
> not tied to key properties

where does this restriction come from?

**Comment by KavithaSiva** [comment]
> > not tied to key properties
> 
> where does this restriction come from?

We don't have a restriction in code, but those are the only fields that can be excluded from a PUT request. I read this in the [specification](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_UpdateanEntity).

**Comment by KavithaSiva** [comment]
I am getting `vale` spelling errors on `approuter` and `kubeconfig` in Kyma do you know how to make vale ignore these words? @MatKuhr

**Comment by MatKuhr** [comment]
where do you read this? The specification explicitly mentions:

> Key and other properties marked as read-only in metadata (including computed properties), as well as dependent properties that are not tied to key properties of the principal entity, can be omitted from the request. 

----

You can list accepted words [here](https://github.com/SAP/cloud-sdk/blob/main/styles/Vocab/SAP/accept.txt).

**Comment by KavithaSiva** [comment]
> where do you read this? The specification explicitly mentions:
> 
> > Key and other properties marked as read-only in metadata (including computed properties), as well as dependent properties that are not tied to key properties of the principal entity, can be omitted from the request.

Doesn't this mean that only these properties can be excluded from the PUT request? Or am I missing something very obvious here?

**Comment by MatKuhr** [comment]
Sorry, I misread this like so:

> excluding (read-only fields or dependent properties) that are not tied to key properties

but what is meant is apparently

>  excluding read-only fields (which may be key properties) or (dependent properties that are not tied to key properties)

Not sure how to word it more clearly though. Maybe just copy the sentence from the OData docs 🤷🏻‍♂️

File: docs/java/features/odata/type-safe-client-odata-v4.mdx [hunk 1]
```diff
@@ -537,7 +538,7 @@ service.updateBusinessPartner(partner)
     //method call optional, since active by default
     .modifyingEntity()
     //add the Business Partner Full Name explicitly to the update request
```
Comment (MatKuhr) [inline docs/java/features/odata/type-safe-client-odata-v4.mdx:540]:
```suggestion
    //add the Business Partner full name and address explicitly to the update request
```

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 2]
```diff
@@ -671,7 +672,7 @@ service.updateBusinessPartner(partner)
     //method call optional, since active by default
     .modifyingEntity()
     //add the Business Partner Full Name explicitly to the update request
```
Comment (MatKuhr) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:674]:
```suggestion
    //add the Business Partner full name and address explicitly to the update request
```

**Review by MatKuhr — APPROVED** [review]
minor code improvement, otherwise lgtm

**Pull request merged**